Nitpicking abi-spec.rst

pirapira opened the pull request.

pirapira committed:
Nitpicking abi-spec.rst

axic commented on docs/abi-spec.rst:
```diff
@@ -6,14 +6,14 @@
 The Application Binary Interface is the standard way to interact with contracts in the Ethereum ecosystem, both
 from outside the blockchain and for contract-to-contract interaction. Data is encoded following its type,
-according to this specification.
+according to this specification.  The types are not encoded.
```
What do you mean by `types are not encoded`?

pirapira replied:
No bytes in the call data say `uint8`.

axic replied:
Wouldn't `The encoding is not self describing and thus requires a schema in order to decode.` be more descriptive?

pirapira replied:
I like that.

axic commented on docs/abi-spec.rst:
```diff
@@ -334,10 +335,6 @@ would result in the JSON:
   }, {
-  "type":"event",
-  "inputs": [{"name":"a","type":"uint256","indexed":true},{"name":"b","type":"bytes32","indexed":false}],
-  "name":"Event2"
```
I see this is duplicated?

pirapira replied:
That's why I'm removing one of the duplicates.

pirapira committed:
Borrow a sentence from @axic

and edit the phrase "following its type", which sounded like the data follows the type in the call data.

chriseth commented on docs/abi-spec.rst:
```diff
@@ -93,6 +93,7 @@ We distinguish static and dynamic types. Static types are encoded in-place and d
 * `string`
 * `T[]` for any `T`
 * `T[k]` for any dynamic `T` and any `k > 0`
+* `(T1,...,Tk)` if any `Ti` is dynamic for `1 <= i <= k`
```
Good catch!
